File: Day 1 - Arrays and Two Pointers/01_TwoSum_BForce/01_TwoSum_BForce.cpp
#include<bits/stdc++.h>
using namespace std;


class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int length = nums.size();
        vector<int> result;
        for (int i =0; i < length - 1; i++){
            for (int j = i+1; j < length; j++) {
                if (nums[i] + nums[j] == target) {
                    result.push_back(i);
                    result.push_back(j);
                    return result;
                }
            }
        }
        return result;
    }
};

int main() {
    Solution s;
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
    vector<int> result = s.twoSum(nums, target);
    for (int x: result) {
        cout << x << ' ';
    }
    cout << endl;
    return 0;
}

File: Day 1 - Arrays and Two Pointers/01_TwoSum_BForce.js
// https://leetcode.com/problems/two-sum/

function twoSum(nums, target) {
    let n = nums.length;
    for (let i = 0; i < n -1; i++) {
        for (let j = i + 1; j < n; j++) {
            if (nums[i] + nums[j] == target) {
                return [i,j];
            }
        }
    }
}

console.log(twoSum([2,7,11,15], 9));